app/apptesting/test_helpers.go

```diff
@@ -263,15 +263,23 @@ func CopyConnectionAndClientToPath(path *ibctesting.Path, pathToCopy *ibctesting
 	return path
 }
 
-func  (s *AppTestHelper) ICAPacketAcknowledgement(msgs []sdk.Msg) channeltypes.Acknowledgement {
+func (s *AppTestHelper) ICAPacketAcknowledgement(msgs []sdk.Msg, msgResponse *proto.Message) channeltypes.Acknowledgement {
 	txMsgData := &sdk.TxMsgData{
 		Data: make([]*sdk.MsgData, len(msgs)),
 	}
 	for i, msg := range msgs {
-		msgResponse := []byte("msg_response")
+		var data []byte
+		var err error
+		if msgResponse != nil {
+			// see: https://github.com/cosmos/cosmos-sdk/blob/1dee068932d32ba2a87ba67fc399ae96203ec76d/types/result.go#L246
+			data, err = proto.Marshal(*msgResponse)
+			s.Require().NoError(err, "marshal error")
+		} else {
+			data = []byte("msg_response")
+		}
```

Comment:
do you think it'd make sense to generalize this to accept a list of msgResponses? Then you could leverage it [here](https://github.com/Stride-Labs/stride/pull/234/files#diff-b534601426c987137850288cf1ff0c31eec9ba0127489956e0399974b15dcf33R273-R292)

Maybe something like:
```go
func (s *AppTestHelper) ICAPacketAcknowledgement(msgs []sdk.Msg, msgResponses []*proto.Message) channeltypes.Acknowledgement {
	txMsgData := &sdk.TxMsgData{
		Data: make([]*sdk.MsgData, len(msgs)),
	}
	for i, msg := range msgs {
		var data []byte
		var err error
		if len(msgResponses) != 0 {
			// see: https://github.com/cosmos/cosmos-sdk/blob/1dee068932d32ba2a87ba67fc399ae96203ec76d/types/result.go#L246
                        resp := msgResponses[len(msgResponses) % i]
			data, err = proto.Marshal(*resp)
			s.Require().NoError(err, "marshal error")
		} else {
			data = []byte("msg_response")
		}
               ...
```

That way it would still support using a mock response:
```go
ICAPacketAcknowledgement(msgs, {emptyList}) 
```
And or using the same response for each message
```go
ICAPacketAcknowledgement(msgs, {listWithOneElement}) 
```

Reply:
I think the most robust solution would be to add a struct like
```
type ICAPacketAcknowledgement struct {
  msg sdk.Msg
  msgResponse proto.Message
}
```

given we don't need this anywhere today, going to leave it out - but we might need to add it soon if we write more complex tests!

Reply:
Sweet - I'm also very much in the camp of not trying to overgeneralize things before the use cases are known!